lib/presentation/widgets/list_characters.dart

```diff
@@ -0,0 +1,60 @@
+import 'package:flutter/material.dart';
+import '../../data/model/response_models/character.dart';
+import '../navigation/app_routes.dart';
+
+class ListCharacters extends StatelessWidget {
+  final List<Character>? list;
+
+  const ListCharacters({Key? key, required this.list}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.all(6.0),
+      child: ListView.builder(
+        itemCount: list?.length,
+        itemBuilder: (context, index) {
+          final item = list?[index] as Character;
+          return Card(
+            shape: const RoundedRectangleBorder(
+              borderRadius: BorderRadius.all(
+                Radius.circular(5.0),
+              ),
+            ),
+            child: InkWell(
+              onTap: () {
+                Navigator.pushNamed(context, AppRoutes.detailScreen,
+                    arguments: item.id);
+              },
+              child: Row(
+                children: [
+                  ClipRRect(
+                    borderRadius: const BorderRadius.only(
+                      topLeft: Radius.circular(5.0),
+                      bottomLeft: Radius.circular(5.0),
+                    ),
+                    child: FadeInImage.assetNetwork(
+                      placeholder: 'assets/graphics/placeholder-150x150.png',
+                      image:
+                          "${item.thumbnailPath}/standard_medium.${item.thumbnailExtension}",
```

Comment:
Please extract such things into methods like `_getThumbnailUrl(item)` or even better create a calculated value in `Character` class: `String get thumbnailUrl => "$thumbnailPath/standard_medium.$thumbnailExtension";`
And use it here like this: `item.thumbnailUrl`.

Reply:
add method in Character class and Series class. And all non-null chek in Detail and ListCharacter screens.